doc/source/data/key-concepts.rst

```diff
@@ -146,6 +146,16 @@ scheduling strategy for all Datasets tasks/actors, using the global
   :start-after: __resource_allocation_begin__
   :end-before: __resource_allocation_end__
 
+Fault tolerance
+===============
+
+Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>` in Ray core. Specifically, if the driver that created a ``Dataset`` is still alive, then Ray core will guarantee that the underlying data is still accessible. This works through **lineage reconstruction**: a Dataset is a collection of Ray objects stored in shared memory, and if any of these objects are lost, then Ray will recreate these by re-executing the task(s) that created them.
```

Comment:
IIUC, any worker can be the process that creates a dataset?

Reply:
Ah yeah, technically any worker can do it but I think it's usually the driver that will create it. I wanted to put driver here to make it clear that we're talking about the worker that invokes Dataset transforms, not the worker that executes them. I can change this to "worker" though.